agroal/src/main/java/org/wildfly/extension/agroal/deployment/DataSourceDefinitionInjectionSource.java

```diff
@@ -0,0 +1,238 @@
+        }
+        if ( user != null && !user.isEmpty() ) {
+            connectionFactoryConfiguration.principal( new NamePrincipal( user ) );
+        }
+        if ( password != null && !password.isEmpty() ) {
+            connectionFactoryConfiguration.credential( new SimplePassword( password ) );
+        }
+        connectionFactoryConfiguration.jdbcTransactionIsolation( AgroalConnectionFactoryConfiguration.TransactionIsolation.fromLevel( isolationLevel ) );
+
+        AgroalConnectionPoolConfigurationSupplier connectionPoolConfiguration = new AgroalConnectionPoolConfigurationSupplier();
+        connectionPoolConfiguration.connectionFactoryConfiguration( connectionFactoryConfiguration );
+
+        if ( initialPoolSize >= 0 ) {
+            connectionPoolConfiguration.initialSize( initialPoolSize );
+        }
+        if ( minPoolSize >= 0 ) {
+            connectionPoolConfiguration.minSize( minPoolSize );
+        }
+        if ( maxPoolSize >= 0 ) {
+            connectionPoolConfiguration.maxSize( maxPoolSize );
+        }
+        if ( maxIdleTime >= 0 ) {
+            connectionPoolConfiguration.reapTimeout( Duration.ofSeconds( maxIdleTime ) );
+        }
+
+        AgroalDataSourceConfigurationSupplier dataSourceConfiguration = new AgroalDataSourceConfigurationSupplier();
+        dataSourceConfiguration.connectionPoolConfiguration( connectionPoolConfiguration );
+
+        ContextNames.BindInfo bindInfo = ContextNames.bindInfoForEnvEntry( context.getApplicationName(), context.getModuleName(), context.getComponentName(), !context.isCompUsesModule(), jndiName );
+        ServiceName dataSourceServiceName = DATASOURCE_DEFINITION_SERVICE_PREFIX.append( bindInfo.getBinderServiceName().getCanonicalName() );
+
+        // This is the service responsible for the JNDI binding, with a dependency on the datasource service that acts as a ManagedReferenceFactory and is used as the injection source
+        BinderService binderService = new BinderService( bindInfo.getBindName(), this );
+        ServiceBuilder<?> binderBuilder = phaseContext.getServiceTarget().addService( bindInfo.getBinderServiceName(), binderService )
+                .addDependency( dataSourceServiceName, ManagedReferenceFactory.class, binderService.getManagedObjectInjector() )
+                .addDependency( bindInfo.getParentContextServiceName(), ServiceBasedNamingStore.class, binderService.getNamingStoreInjector() );
+        binderBuilder.install();
+
+        DataSourceDefinitionService dataSourceService = new DataSourceDefinitionService( bindInfo, dataSourceConfiguration );
+        ServiceBuilder<ManagedReferenceFactory> dataSourceServiceBuilder = phaseContext.getServiceTarget().addService( dataSourceServiceName, dataSourceService );
+
+        if ( transactional ) {
+            dataSourceServiceBuilder.addDependency( TxnServices.JBOSS_TXN_TRANSACTION_MANAGER, TransactionManager.class, dataSourceService.getTransactionManagerInjector() );
```

Comment:
These services are deprecated.  For now it's appropriate to add a no-value dependency just on the ``JBOSS_TXN_TRANSACTION_MANAGER`` service, but otherwise just call ``ContextTransactionManager.getInstance()`` to get the transaction manager.  Similar for ``ContextTransactionSynchronizationRegistry``.